src/index.js

```diff
@@ -8,26 +8,35 @@ const inputElement = document.getElementById("input-field");
 const submitButtonElement = document.getElementById("submit-button");
 const submitMessageElement = document.getElementById("submit-message");
 
-submitButtonElement.onclick = () => {
+const handleWidgetEnd = () => {
   widgetSDK.sendUserData({
     responseData: inputElement.value
   }, (event) => {
     if (event.type === "SEND_USER_DATA_SUCCESS") {
-      submitMessageElement.innerText = "Data was successfully submitted";
+      submitMessageElement.innerText = "Data was successfully submitted. The widget will close in a moment.";
       submitButtonElement.disabled = true;
     } else {
-      submitMessageElement.innerText = "Data submission failed, please try again";
+      submitMessageElement.innerText = "Data submission failed, please try again.";
     }
   });
-};
+  setTimeout(() => {
+    widgetSDK.setWidgetClose();
+  }, 2.5 * 1000);
```

Comment:
Magic numbers. Not a blocker.

Reply:
Good callout. I'll add a comment about the 2.5 seconds to milliseconds in another, less urgent PR in the upcoming weeks. Or I'll add an input field so the bot builder can configure the number of seconds.